llvm/cmake/modules/GetHostTriple.cmake

```diff
@@ -2,7 +2,7 @@
 # Invokes config.guess
 
 function( get_host_triple var )
-  if( MSVC )
+  if( MSVC OR (CMAKE_SYSTEM_NAME STREQUAL "Windows" AND CMAKE_C_COMPILER_ID MATCHES "Clang" AND NOT MINGW AND NOT MSYS))
```

Comment:
IMO, this if condition should be organized something like following:
```cmake
if(CMAKE_SYSTEM_NAME STREQUAL "Windows")
    if(CMAKE_C_COMPILER_ID MATCHES "Clang")
        if(MSVC)
            message(STATUS "Clang is operating in MSVC mode (MSVC style driver: clang-cl).")
        elseif(CMAKE_C_COMPILER_ARCHITECTURE_ID)
            message(STATUS "Clang is operating in MSVC mode (GNU style driver).")
        else()
            message(STATUS "Clang MinGW mode")
        endif()
    elseif(MSVC)
        message(STATUS "Compiler is MSVC (cl compiler).")
endif()

Reply:
Yeah, something like that would certainly be cleaner - but that would probably also be a much larger refactoring of the cmake code here.

Anyway, I'll accept the change here as I think this version should function correctly, but I don't mind such a restructuring either (maybe ideally as a separate step?).

Reply:
I'll take care of that later, just first want to be sure all changes there to run all test, build everything fine. Not sure how much other changes needed... looking at it.